src/Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -690,33 +691,42 @@ public bool TryParseInterface(string intf, [NotNullWhen(true)] out IReadOnlyList
     }
 
     /// <inheritdoc/>
-    public bool HasRemoteAccess(IPAddress remoteIP)
+    public RemoteAccessPolicyResult ShouldAllowServerAccess(IPAddress remoteIP)
     {
         var config = _configurationManager.GetNetworkConfiguration();
+        if (IsInLocalNetwork(remoteIP))
+        {
+            return RemoteAccessPolicyResult.Allow;
+        }
+
         if (config.EnableRemoteAccess)
         {
+            if (!_remoteAddressFilter.Any())
+            {
+                // No filter on remote addresses, allow any of them.
+                return RemoteAccessPolicyResult.Allow;
+            }
+
             // Comma separated list of IP addresses or IP/netmask entries for networks that will be allowed to connect remotely.
             // If left blank, all remote addresses will be allowed.
-            if (_remoteAddressFilter.Any() && !IsInLocalNetwork(remoteIP))
-            {
-                // remoteAddressFilter is a whitelist or blacklist.
-                var matches = _remoteAddressFilter.Count(remoteNetwork => NetworkUtils.SubnetContainsAddress(remoteNetwork, remoteIP));
-                if ((!config.IsRemoteIPFilterBlacklist && matches > 0)
-                    || (config.IsRemoteIPFilterBlacklist && matches == 0))
-                {
-                    return true;
-                }
 
-                return false;
+            // remoteAddressFilter is a whitelist or blacklist.
+            var matches = _remoteAddressFilter.Count(remoteNetwork => NetworkUtils.SubnetContainsAddress(remoteNetwork, remoteIP));
```

Comment:
Since we only care about 0/1 we can do `.Any()` here instead for a small fast-path